src/irmin-pack/unix/chunked_suffix.ml

```diff
@@ -98,17 +112,58 @@ module Make (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
     let close t =
       (* Close immutable chunks, ignoring errors. *)
       let _ =
-        Array.sub t 0 (Array.length t - 1)
+        Array.sub t.chunks 0 (Array.length t.chunks - 1)
         |> Array.iter @@ fun chunk ->
            let _ = Ao.close chunk.ao in
            ()
       in
       (* Close appendable chunk and keep error since this
          is the one that can have a pending flush. *)
       (appendable t).ao |> Ao.close
+
+    let wrap_error result =
+      Result.map_error
+        (fun err -> (err : add_new_error :> [> add_new_error ]))
+        result
+
+    let reopen_last_chunk ~open_chunk t =
+      (* Close the previous appendable chunk and reopen as non-appendable. *)
+      let open Result_syntax in
+      let ({ idx; ao; suffix_off } as last_chunk) = appendable t in
+      let is_legacy = idx = 0 in
```

Comment:
Now that we have two instances of this calculation, wdyt about making it a small helper function that both this and `open_` can use?

Reply:
I assumed you meant the `Int63.Syntax.(suffix_off + chunk_len)` line, done.

Reply:
Oh, sorry, I meant the `let is_legacy = idx = 0` calculation (just so we centralize the notion of the legacy chunk being the one with idx = 0), but that's a good one too. :smile: